HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,706 @@
+
+This HIP changes this behavior in the following ways:
+
+- The value `-1` is now permitted for `max_auto_associations`, and this value means “unlimited”.
+- The account holder no longer prepays for unused slots regardless of the setting for `max_auto_associations`. Slots are only paid for when used, and are initially paid for by the sender during automatic associations. The account is responsible for covering rent for each association after the first auto-renewal period. `used_auto_associations` is incremented every time an auto-association is made.
+- The default value for automatically-created accounts will now be `-1`. This means, if an account is created automatically (by performing a token transfer to an alias that does not yet exist, see [HIP-583](https://hips.hedera.com/hip/hip-583)), then the automatically created account is configured with unlimited automatic token associations.
+
+### Crypto Transfers
+
+Normal crypto-transfers remain the same as today, with one difference. If Account A sends a token to Account B, and Account B is not associated with that token but has an available auto-association slot, then Account A will pay for that association and the first auto-renewal period’s rent, in addition to the typical transfer costs.
+
+![Crypto Transfer Transaction](../assets/hip-904/frictionless-airdrop-cryptoTransferTransaction.png)
+
+### Airdrops
+
+Airdrops are similar to crypto transfers, but differ in one fundamental way - when an airdrop is performed and the receiver does not have available or automatic association slots, rather than failing, the transfer will be kept in state as a *pending transfer*. To represent this new semantic, there will be a new `TokenAirdropTransaction` meant for use specifically by those attempting an airdrop, with the following semantics:
+
+![Token Airdrop Transaction](../assets/hip-904/frictionless-airdrop-tokenAirdropTransaction.png)
+
+> ⚠️ `TokenAirdropTransaction` will not support tokens with custom fallback fees that require the receiver to pay the fallback fee.
```

Comment:
what about fee key then?  custom fee could be changed between time of airdrop and acceptance.

Reply:
Good question.
Fees to payer would be assessed at the time of the TokenAirdrop transaction. So whatever custom fees are then, doesn't matter what it gets changed to

Reply:
Note this is called out here https://github.com/hashgraph/hedera-improvement-proposal/pull/904/files#diff-da073d1b1241c0622fd8b02747d2d95afcbd719a2f61eb953bc9df79d64fa336R102

Reply:
glad this was explicitly called out. Also thought that fallback fee should be *paid by the sender*.